required.go

```diff
@@ -6,35 +6,73 @@ import (
 	"strings"
 )
 
+type Option uint8
+
+const (
+	// Allows empty string, array or map. Default is false
+	AllowEmptyString Option = iota
+	AllowEmptyArray
+	AllowEmptyMap
+	AllowEmptySlice
+)
+
 // Required Validator is very simple
-type RequiredValidator struct{}
+type Required struct {
+	opts map[Option]bool
+}
 
-// IsValid determines if the given value satisfies a "Required" rule.
-func (r *RequiredValidator) IsValid(value interface{}) (bool, []error) {
+func (r *Required) SetOpts(opts map[Option]bool) {
+	r.opts = opts
+}
 
+// IsValid determines if the given value satisfies a "Required" rule.
+func (r *Required) IsValid(value interface{}) (bool, []error) {
 	if value == nil {
-		return false, singleErrorSlice("nil is not a valid value for RequiredValidator")
+		return false, singleErrorSlice("nil is not a valid value for Required")
```

Comment:
I would make this error message more in line with what the others are like, and in fact I would bake the english into the helper function - we should be able to call here, for instance: `singleErrorSlice("Base-level", nil, "Required")`
and it would output:
`Base-level value(nil) is not a valid value for Required`
and below
`String value("") is not a valid value for Required` etc

However I'm not sure that the english there is quite right either, but this would give us some freedom to mess around with it and not have to worry about updating it all over the place